healpix_alchemy/point.py

```diff
@@ -1,25 +1,57 @@
 """Spatial indexing for astronomical point coordinates."""
+import numpy as np
 from sqlalchemy.ext.declarative import declared_attr
 from sqlalchemy.ext.hybrid import hybrid_method, hybrid_property
 from sqlalchemy.schema import Column, Index
-from sqlalchemy.sql import and_
+from sqlalchemy.sql import and_, between
 from sqlalchemy.types import Float
 
-from .math import sind, cosd
-
 __all__ = ('Point',)
 
 
+def get_x_for_ra_dec(ra, dec):
+    return np.cos(np.deg2rad(ra)) * np.cos(np.deg2rad(dec))
+
+
+def get_y_for_ra_dec(ra, dec):
+    return np.sin(np.deg2rad(ra)) * np.cos(np.deg2rad(dec))
+
+
+def get_z_for_ra_dec(ra, dec):
+    return np.sin(np.deg2rad(dec))
+
+
+def get_x_for_context(context):
+    params = context.get_current_parameters()
+    return get_x_for_ra_dec(params['ra'], params['dec'])
+
+
+def get_y_for_context(context):
+    params = context.get_current_parameters()
+    return get_y_for_ra_dec(params['ra'], params['dec'])
+
+
+def get_z_for_context(context):
+    params = context.get_current_parameters()
+    return get_z_for_ra_dec(params['ra'], params['dec'])
+
+
 class Point:
     """Mixin class to add a point to a an SQLAlchemy declarative model."""
 
     def __init__(self, *args, ra=None, dec=None, **kwargs):
         super().__init__(*args, **kwargs)
         self.ra = ra
         self.dec = dec
+        self.x = get_x_for_ra_dec(ra, dec)
```

Comment:
Adding these three columns increases the size of the table by >2x, perhaps the backend should just store xyz and ra, dec should be implemented as python only properties?

Storing both xyz and ra,dec seems like overkill - shouldn't be necessary to store both.

Reply:
It don't think it actually increases the table size because those values were calculate for the index anyway.

If we remove the ra and dec columns, then we may have to calculate them on the fly when displaying these values.

Reply:
Any way to test this? Can we measure the size of a table + index in both scenarios?

Reply:
The unit test report now includes total tables + indices size for each unit test. The impact of this patch is very small.

## On main branch

```
================================ database size =================================
healpix_alchemy/tests/test_point.py::test_cross_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cross_join[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cross_join[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cross_join[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_self_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_self_join[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_self_join[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_self_join[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10000]: 4232 kB
healpix_alchemy/tests/test_regions.py::test_fields: 1432 kB
```

## With this PR

```
================================ database size =================================
healpix_alchemy/tests/test_point.py::test_cross_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cross_join[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cross_join[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cross_join[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_self_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_self_join[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_self_join[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_self_join[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cone_search[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10000]: 4712 kB
healpix_alchemy/tests/test_regions.py::test_fields: 1432 kB
```